src/gui/csvImport/CsvImportWidget.cpp

```diff
@@ -131,15 +132,27 @@ void CsvImportWidget::updateTableview() {
 
 void CsvImportWidget::updatePreview() {
 
+    int minSkip = 0;
+    if (m_ui->checkBoxFieldNames->isChecked())
+        minSkip = 1;
     m_ui->labelSizeRowsCols->setText(m_parserModel->getFileInfo());
-    m_ui->spinBoxSkip->setValue(0);
-    m_ui->spinBoxSkip->setMaximum(qMax(0, m_parserModel->rowCount() - 1));
+    m_ui->spinBoxSkip->setValue(minSkip);
```

Comment:
Just a minor note:
here you can use [setRange](https://doc.qt.io/qt-5/qspinbox.html#setRange) to both set max and min value.
You should set the min value to `minSkip` so when the `checkBoxFieldNames` is selected, people can't skip the first row in the spinbox